src/Integrations/Integrations/Swoole/SwooleIntegration.php

```diff
@@ -97,21 +97,14 @@ function (HookData $hook) use (&$rootSpan, $integration) {
                 $rootSpan->resource = $request->server['request_method'] . ' ' . $normalizedPath;
                 $rootSpan->meta[Tag::HTTP_METHOD] = $request->server['request_method'];
 
-                $serverProtocol = $request->server['server_protocol'] ?? 'HTTP/1.1';
-                $scheme = strpos($serverProtocol, 'HTTPS') !== false ? 'https://' : 'http://';
+                $scheme = $server->ssl ? 'https://' : 'http://';
```

Comment:
Can you move that comparison before the install_hook call? and use ($scheme) instead? Apart from the minuscule micro-optimization, it also avoids having a circular dependency. Not sure if it makes any difference, but I think we should.